src/lib/parseParams.ts

```diff
@@ -0,0 +1,200 @@
+        const configs = [".js", ".cjs"];
+        // in order of precedence
+        return [packageJson].concat(
+            rcs.concat(configs).map(extension => `.${denoify}rc${extension}`),
+            configs.map(extension => `${denoify}.config${extension}`)
+        );
+    })();
+
+    function parseConfig({ file, config }: { file: string | undefined; config: string | undefined }): ConfigFileType {
+        if (file === undefined || config === undefined) {
+            return {
+                type: "absent"
+            };
+        }
+        if (file.endsWith(".json")) {
+            return {
+                type: "json",
+                file,
+                config
+            };
+        } else if (file.split(".").length === 2) {
+            return {
+                type: "extensionless",
+                file,
+                config
+            };
+        } else if (file.endsWith(".cjs") || file.endsWith(".js")) {
+            return {
+                type: "js",
+                file,
+                config
+            };
+        } else if (file.endsWith(".yaml") || file.endsWith(".yml")) {
+            return {
+                type: "yaml",
+                file,
+                config
+            };
+        } else {
+            return {
+                type: "absent"
+            };
+        }
+    }
+
+    return {
+        getFileTypeAndContent(content: (file: string) => Promise<string | undefined>) {
+            return supportedConfigFile.reduce(async (configFileType, file) => {
+                if ((await configFileType).type !== "absent") {
+                    return configFileType;
+                } else {
+                    const config = await content(file);
+                    if (!config) {
+                        return configFileType;
+                    } else if (file !== packageJson) {
+                        return parseConfig({
+                            file,
+                            config
+                        });
+                    } else if (!YAML.parse(config).denoify) {
+                        return configFileType;
+                    } else {
+                        return parseConfig({
+                            file,
+                            config
+                        });
+                    }
+                }
+            }, Promise.resolve({ type: "absent" }) as Promise<ConfigFileType>);
+        },
+        parseAsDenoifyConfig(configFileType: ConfigFileType) {
+            switch (configFileType.type) {
+                case "absent":
+                    return undefined;
+                case "json":
+                case "yaml":
+                case "extensionless":
+                    // yaml is a superset of json
+                    const parsed = YAML.parse(configFileType.config);
+                    return parseAsDenoifyParams(configFileType.file !== packageJson ? parsed : parsed.denoify);
```

Comment:
```suggestion
                case "extensionless": {
                    // yaml is a superset of json
                    const parsed = YAML.parse(configFileType.config);
                    return parseAsDenoifyParams(configFileType.file !== packageJson ? parsed : parsed.denoify);
                }
```

Use anonymous block for scopping declared variables.